### PR TITLE
acl: Add a sample script and bug fixes

### PR DESCRIPTION
**Pull request by changlan**

### CODE REVIEW

**Comment by justinemarie** [comment]
Thanks for adding the sample script! This is super helpful.

File: core/modules/acl.h [hunk 1]
```diff
@@ -15,10 +15,10 @@ class ACL final : public Module {
   struct ACLRule {
     bool Match(IPAddress sip, IPAddress dip, uint16_t sport,
                uint16_t dport) const {
-      return ((src_ip.addr != 0 && src_ip.Match(sip)) &&
-              (dst_ip.addr != 0 && dst_ip.Match(dip)) &&
-              (src_port != 0 && src_port == sport) &&
-              (dst_port != 0 && dst_port == dport));
+      return (src_ip.addr == 0 || (src_ip.addr != 0 && src_ip.Match(sip))) &&
```
Comment (sangjinhan) [inline core/modules/acl.h]: The expression is somewhat redundant? `(src_ip.addr == 0 || src_ip.Match(sip)) && ...` should suffice. Or even better, `src_ip.Match(sip) && ...` alone should be enough.

File: bessctl/conf/samples/acl.bess [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+packets = [gen_packet(scapy.UDP, '172.16.100.1', '10.0.0.1'),
+           gen_packet(scapy.UDP, '172.12.55.99', '12.34.56.78'),
+           gen_packet(scapy.UDP, '172.12.55.99', '10.0.0.1'),
+           gen_packet(scapy.UDP, '172.16.100.1', '12.34.56.78'),
+           gen_packet(scapy.TCP, '172.12.55.99', '12.34.56.78'),
+           gen_packet(scapy.UDP, '192.168.1.123', '12.34.56.78'),
+          ]
+
+fw::ACL(rules=[{'src_ip': '172.12.1.1/16', 'drop': False}])
```
Comment (sangjinhan) [inline bessctl/conf/samples/acl.bess]: `172.16.0.0/16`?

**Comment by sangjinhan** [comment]
Merged, thanks!

**Pull request merged**